File: Java/labs/lab04_Builder/task_3_1/src/Wheel.java
public class Wheel {
    public enum Material {
        Steel, //  стальний диск
        Alloy,  //  легкосплавний
        Forged  //  кований легкосплавний диск
    }

    final private Material material;
    final int diameter;

    public Wheel(Material material, int diameter) {
        this.material = material;
        this.diameter = diameter;
    }

    @Override
    public String toString() {
        return "Wheel{" +
                "material=" + material +
                ", diameter=" + diameter +
                '}';
    }
}

File: Cpp/lab04_Builder/task_3_1/src/wheel.cpp
#include "wheel.hpp"


Wheel::Wheel(Wheel::Material material, int diameter)
: material(material), diameter(diameter) {}

std::ostream& operator<<(std::ostream& os, Wheel const& wheel)
{
  os << "Wheel{material=";

  switch (wheel.material)
  {
    case Wheel::Material::Steel:
      os << "Steel";
      break;
    case Wheel::Material::Alloy:
      os << "Alloy";
      break;
    case Wheel::Material::Forged:
      os << "Forged";
      break;
  }

  os << ", diameter=" << wheel.diameter << "}";

  return os;
}
